chalice/deploy/planner.py

```diff
@@ -279,14 +290,100 @@ def plan_scheduledevent(self, resource):
         ]
         return plan
 
+    def plan_restapi(self, resource):
+        # type: (models.RestAPI) -> List[models.Instruction]
+        function = resource.lambda_function
+        function_name = function.function_name
+        varname = '%s_lambda_arn' % function.resource_name
+        lambda_arn_var = Variable(varname)
+        shared_plan = [
+            # The various API gateway API calls need
+            # to know the region name and account id so
+            # we'll take care of that up front and store
+            # them in variables.
+            models.BuiltinFunction(
+                'parse_arn',
+                [lambda_arn_var],
+            ),
+            models.StoreValue('parsed_lambda_arn'),
+            models.JPSearch('account_id'),
+            models.StoreValue('account_id'),
+            models.LoadValue('parsed_lambda_arn'),
+            models.JPSearch('region'),
+            models.StoreValue('region_name'),
+            # The swagger doc uses the 'api_handler_lambda_arn'
+            # var name so we need to make sure we populate this variable
+            # before importing the rest API.
+            models.LoadValue(varname),
+            models.StoreValue('api_handler_lambda_arn'),
+        ]
+        if not self._remote_state.resource_exists(resource):
+            plan = shared_plan + [
+                models.APICall(
+                    method_name='import_rest_api',
+                    params={'swagger_document': resource.swagger_doc},
+                ),
+                # We get rest_api_id as the return value, and need this value
+                # in subsequent API calls.
+                models.StoreValue(name='rest_api_id'),
+                models.RecordResourceVariable(
+                    resource_type='rest_api',
+                    resource_name=resource.resource_name,
+                    name='rest_api_id',
+                    variable_name='rest_api_id',
+                ),
+                models.APICall(
+                    method_name='deploy_rest_api',
+                    params={'rest_api_id': Variable('rest_api_id'),
+                            'api_gateway_stage': resource.api_gateway_stage},
+                ),
+                models.APICall(
+                    method_name='add_permission_for_apigateway_if_needed',
+                    params={'function_name': function_name,
+                            'region_name': Variable('region_name'),
+                            'account_id': Variable('account_id'),
+                            'rest_api_id': Variable('rest_api_id')},
+                ),
+            ]
+        else:
+            deployed = self._remote_state.resource_deployed_values(resource)
+            plan = shared_plan + [
+                models.Push(deployed['rest_api_id']),
```

Comment:
Capturing some offline feedback, I'd be curious to see what these would look like if there was only a stack or only variables.